### PR TITLE
Configure pushers

### PR DESCRIPTION
**Pull request by gnunicorn**
Build on top of #945 this allows you to configure and remove any particular pusher via the account settings. Additionally this contains the features to add email-pushers but that isn't yet activated until #926 is through.

Screenshot_1697732716
Screenshot_1697732718

Best to review after #945 has been merged I suppose.

### CODE REVIEW

**Comment by bitfriend** [comment]
`flutter_secure_storage` supports `isCupertinoProtectedDataAvailable()` in only ios and macos.
It is not supported in windows.
image
App compiling failed in windows.

**Comment by bitfriend** [comment]
@gnunicorn You can change `/app/packages/rust_sdk/pubspec.yaml` like `/app/pubspec.yaml`.
You already used `develop` branch not `main` branch of `flutter_secure_storage` in `/app/pubspec.yaml`.

**Review by bitfriend — APPROVED** [review]
Great work.
I read your tips about linux build of `flutter_secure_storage`.
Thank you for your kindly explanation.

File: app/lib/features/settings/pages/notifications_page.dart [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+  }
+
+  SettingsSectionWithTitleActions _pushTargets(
+    BuildContext context,
+    WidgetRef ref,
+  ) {
+    return SettingsSectionWithTitleActions(
+      title: const Text('Notification Targets'),
+      actions: const [
+        // FIXME: enable once we have 3pid support
```
Comment (gtalha07) [inline app/lib/features/settings/pages/notifications_page.dart:134]: is this commented block to be enabled in consequent PR's ?

**Review by gtalha07 — APPROVED** [review]
LGTM!

**Pull request merged**